crates/grafbase-hooks/src/hooks/authorization/edge_node_post_execution_arguments.rs

```diff
@@ -0,0 +1,182 @@
+    nodes: Vec<String>,
+    metadata: String,
+}
+
+impl EdgeNodePostExecutionArguments {
+    pub(crate) fn new(definition: crate::wit::EdgeDefinition, nodes: Vec<String>, metadata: String) -> Self {
+        Self {
+            definition,
+            nodes,
+            metadata,
+        }
+    }
+
+    /// The name of the parent type of the edge.
+    ///
+    /// For the following GraphQL schema:
+    ///
+    /// ```graphql
+    /// type User {
+    ///     id: Int!
+    ///     name: String!
+    /// }
+    ///
+    /// type Query {
+    ///     users: [User!]! @authorized(node: "id")
+    /// }
+    /// ```
+    ///
+    /// And the query:
+    ///
+    /// ```graphql
+    /// query {
+    ///     users { name }
+    /// }
+    /// ```
+    ///
+    /// The parent type name is `Query`.
+    pub fn parent_type_name(&self) -> &str {
+        &self.definition.parent_type_name
+    }
+
+    /// The name of the authorized edge.
+    ///
+    /// For the following GraphQL schema:
+    ///
+    /// ```graphql
+    /// type User {
+    ///     id: Int!
+    ///     name: String!
+    /// }
+    ///
+    /// type Query {
+    ///     users: [User!]! @authorized(node: "id")
+    /// }
+    /// ```
+    ///
+    /// And the query:
+    ///
+    /// ```graphql
+    /// query {
+    ///     users { name }
+    /// }
+    /// ```
+    ///
+    /// The field name is `users`.
+    pub fn field_name(&self) -> &str {
+        &self.definition.field_name
+    }
+
+    /// The nodes of the edge, serialized as a JSON objects.
+    /// This method will deserialize the nodes into either `serde_json::Value` or a custom struct.
+    ///
+    /// For the following GraphQL schema:
+    ///
+    /// ```graphql
+    /// type User {
+    ///     id: Int!
+    ///     name: String!
+    /// }
+    ///
+    /// type Query {
+    ///     users: [User!]! @authorized(node: "id")
+    /// }
+    /// ```
+    ///
+    /// And the query:
+    ///
+    /// ```graphql
+    /// query {
+    ///     users { name }
+    /// }
+    /// ```
+    ///
+    /// The query returns two entities:
+    ///
+    /// ```json
+    /// [
+    ///   {
+    ///     "id": 1,
+    ///     "name": "Alice"
+    ///   },
+    ///   {
+    ///     "id": 2,
+    ///     "name": "Bob"
+    ///   }
+    /// ]
+    /// ```
+    ///
+    /// The arguments can be deserialized into a custom struct like:
+    ///
+    /// ```rust
+    /// #[derive(serde::Deserialize)]
+    /// struct User {
+    ///    id: u64,
+    /// }
+    ///
+    /// # fn foo(arguments: grafbase_hooks::EdgeNodePostExecutionArguments) -> Result<(), serde_json::Error> {
+    /// let parents: Vec<User> = arguments.deserialize_nodes()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// The directive defines the `node` argument as `id`, so the hook gets an object of all
+    /// ids of the returned users.
+    pub fn deserialize_nodes<T: serde::de::DeserializeOwned>(&self) -> Result<Vec<T>, serde_json::Error> {
```

Comment:
would have named those `nodes_as` to write `let x = args.nodes_as::<T>()`  but not even a nitpick, just what comes to mind. Or just `nodes` honestly since today it's the only way to access the data anyway.